script.js

```diff
@@ -1,4 +1,216 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+var scissors = "scissors";
+var stone = "stone";
+var paper = "paper";
+//Keeping track of Computer's and User's wins
+var computerWin = 0;
+var userWin = 0;
+//Keeping track of game mode
+var gameMode = "normal";
+//username
+var username = "";
+
+var main = function (inputSign) {
+  //computer's sign
+  var resultOfComputerSign = computerSign();
+  //Accounting for Gamemode Change
+  if (gameMode == "normal" && inputSign == "reverse") {
+    gameMode = "reverse";
+    var myOutputValue = "Gamemode is now in reverse";
+    return myOutputValue;
+  }
+  if (gameMode == "reverse" && inputSign == "normal") {
+    gameMode = "normal";
+    var myOutputValue = "Gamemode is now normal";
+    return myOutputValue;
+  }
+  //Accounting for Draw scenario
+  if (inputSign == resultOfComputerSign) {
+    //calling special message function
+    var specialMessageHere = specialMessage();
+    //calling gamemode message function
+    var gameModeMessageHere = gameModeMessage();
+    var myOutputValue =
+      "It's a draw " +
+      username +
+      "! Computer selected " +
+      inputSign +
+      "<br> Try again! <br> Computer has so far won " +
+      computerWin +
+      " number of times. <br> You have so far won " +
+      userWin +
+      " number of times. " +
+      specialMessageHere +
+      gameModeMessageHere;
+    return myOutputValue;
+  }
+  //Normal Game mode
+  if (gameMode == "normal") {
+    //Accounting for Lose scenario for Normal Game mode
+    if (
+      (inputSign == "scissors" && resultOfComputerSign == stone) ||
+      (inputSign == "stone" && resultOfComputerSign == paper) ||
+      (inputSign == "paper" && resultOfComputerSign == scissors)
+    ) {
+      //Add to Computer's win
+      computerWin += 1;
+      //calling special message function
+      var specialMessageHere = specialMessage();
+      //calling gamemode message function
+      var gameModeMessageHere = gameModeMessage();
+      var myOutputValue =
+        "Computer selected " +
+        resultOfComputerSign +
+        " you lose! <br> Try again " +
+        username +
+        "! <br> Computer has so far won " +
+        computerWin +
+        " number of times. <br> You have so far won " +
+        userWin +
+        " number of times. " +
+        specialMessageHere +
+        gameModeMessageHere;
+      return myOutputValue;
+    }
```

Comment:
You can consider using JavaScript Template Literals `` instead of ' ' so that you can combine your string values with your variables in one sentence without repeated plus signs. This will save you a lot of time.

https://www.w3schools.com/js/js_string_templates.asp